lib/greenhouse_api/client.rb

```diff
@@ -98,18 +120,20 @@ def data_limit_reached?(data, limit)
       limit && data.length >= limit
     end
 
-    def headers
+    def headers(on_behalf_of = nil)
       credential = Base64.strict_encode64(@api_key + ':')
 
-      {
+      request_header = {
         'Authorization' => 'Basic ' + credential,
       }
+      request_header['On-Behalf-Of'] = on_behalf_of unless on_behalf_of.nil? || on_behalf_of.strip.empty?
```

Comment:
This is a workaround since we don't have access to `present?` here

Reply:
what do you think about actually merging this extra key/val into headers in-line?

so instead of changing this method, the `enable_user` method would be like
```ruby
headers: headers.merge(ON_BEHALF_OF => on_behalf_of_user_id),
```